module/importer/src/Infrastructure/Handler/DeleteSourceCommandHandler.php

```diff
@@ -28,8 +41,19 @@ public function __invoke(DeleteSourceCommand $command): void
     {
         $source = $this->sourceRepository->load($command->getId());
 
-        Assert::notNull($source, sprintf('Can\'t fid source "%s"', $command->getId()->getValue()));
+        Assert::isInstanceOf(
+            $source,
+            AbstractSource::class,
+            sprintf('Can\'t find source "%s"', $command->getId()->getValue())
+        );
+
+        $fileNames = $this->importQuery->getFileNamesBySourceId($source->getId());
 
         $this->sourceRepository->delete($source);
+
+        foreach ($fileNames as $fileName) {
+            $importDeletedCommand = new ImportDeletedCommand($fileName, $source->getType());
```

Comment:
The names of the commanders should tell you what is to happen and not what has happened.

new DeleteImportCommand